website/docs: 2025.8 release notes (#15975)
goauthentik/authentik: website/docs/releases/2025/v2025.8.md

To try out the release candidate, replace your Docker image tag with the latest release candidate number, such as 2025.8.0-rc1. You can find the latest one in [the latest releases on GitHub](https://github.com/goauthentik/authentik/releases). If you don't find any, it means we haven't released one yet.
:::::

## Highlights

- **OAuth2/OpenID Connect back-channel logout**: :ak-preview A server-to-server notification mechanism that allows authentik to notify OAuth2/OpenID providers whenever a user's session is terminated.

- **Event map**: :ak-enterprise View geolocation of user events on a map.

![](../../sys-mgmt/events/event-map-chart.png)

> - **Detailed search**: :ak-enterprise Search for users and events with custom query language to filter on their properties and attributes.

Reviewer: is there any reason why this isn't a relative path

> - **Detailed search**: :ak-enterprise Search for users and events with custom query language to filter on their properties and attributes.

Reviewer: hasn't it always been like that?

> - **Email stage rate limiting**: The email stages can be configured to rate limit the amount of emails sent.

Reviewer 2: ```suggestion
- **Email stage rate limiting**: The email stage can now be configured to rate limit the amount of emails sent.
```

## Breaking changes

### Worker and background tasks revamped

The authentik [worker](../../sys-mgmt/ops/worker.md) and [background tasks](../../sys-mgmt/background-tasks.md) have been reworked for better observability of tasks, and better configurability of scheduled tasks.

This rework also allowed us to not depend on Redis for background tasks. However, we replaced the engine used to manage these tasks, and as such, don't have a seamless migration path.

For instances with a high level of traffic, such as many users logging in, many sign up requests, etc., some tasks may be lost during the upgrade. Instances with low traffic can upgrade during periods of downtime.

To prevent losing tasks during the upgrade, instances with a high level of traffic should follow these instructions:

> - Upgrade the authentik server first.
> - Inspect the old version task queue to check that all tasks are done. Execute the following commands in the not-yet-ugpraded worker container:

Reviewer 2: ```suggestion
1. Start by upgrading the authentik server.
2. Inspect the old version task queue to check that all tasks are done. Execute the following commands in the not-yet-ugpraded worker container:
```

    values={[
        {label: 'docker-compose', value: 'docker-compose'},
        {label: 'Kubernetes', value: 'kubernetes'},
    ]}>
    <TabItem value="docker-compose">

    ```sh
    docker compose exec worker bash -c 'DJANGO_SETTINGS_MODULE=authentik.root.settings celery -A authentik.root.celery inspect active'
    docker compose exec worker bash -c 'DJANGO_SETTINGS_MODULE=authentik.root.settings celery -A authentik.root.celery inspect scheduled'
    docker compose exec worker bash -c 'DJANGO_SETTINGS_MODULE=authentik.root.settings celery -A authentik.root.celery inspect reserved'
    ```

    </TabItem>
    <TabItem value="kubernetes">

    ```sh
    kubectl exec -it deployment/authentik-worker -c worker -- bash -c 'DJANGO_SETTINGS_MODULE=authentik.root.settings celery -A authentik.root.celery inspect active'
    kubectl exec -it deployment/authentik-worker -c worker -- bash -c 'DJANGO_SETTINGS_MODULE=authentik.root.settings celery -A authentik.root.celery inspect scheduled'
    kubectl exec -it deployment/authentik-worker -c worker -- bash -c 'DJANGO_SETTINGS_MODULE=authentik.root.settings celery -A authentik.root.celery inspect reserved'
    ```

    </TabItem>
    </Tabs>

> - Wait for all these commands to report the old task queues as "empty"
> - Finish by upgrading the worker

Reviewer 2: ```suggestion
3. Wait for all these commands to report the old task queues as "empty"
4. Finish by upgrading the worker
```

To try out the release candidate, replace your Docker image tag with the latest release candidate number, such as 2025.8.0-rc1. You can find the latest one in [the latest releases on GitHub](https://github.com/goauthentik/authentik/releases). If you don't find any, it means we haven't released one yet.
:::::

## Highlights

- **OAuth2/OpenID Connect back-channel logout**: :ak-preview A server-to-server notification mechanism that allows authentik to notify OAuth2/OpenID providers whenever a user's session is terminated.

- **Event map**: :ak-enterprise View geolocation of user events on a map.

> ![](../../sys-mgmt/events/event-map-chart.png)

Reviewer: Can we add some alt text here

- **Email stage rate limiting**: The email stage can now be configured to rate limit the amount of emails sent.

## Breaking changes

### Worker and background tasks revamped

The authentik [worker](../../sys-mgmt/ops/worker.md) and [background tasks](../../sys-mgmt/background-tasks.md) have been reworked for better observability of tasks, and better configurability of scheduled tasks.

This rework also allowed us to not depend on Redis for background tasks. However, we replaced the engine used to manage these tasks, and as such, don't have a seamless migration path.

For instances with a high level of traffic, such as many users logging in, many sign up requests, etc., some tasks may be lost during the upgrade. Instances with low traffic can upgrade during periods of downtime.

To prevent losing tasks during the upgrade, instances with a high level of traffic should follow these instructions:

1. Start by upgrading the authentik server.
2. Inspect the old version task queue to check that all tasks are done. Execute the following commands in the not-yet-ugpraded worker container:

> import TabItem from "@theme/TabItem";

Reviewer: don't forget to rename to .mdx

- `authentik_tasks_delayed_in_progress`
- `authentik_tasks_duration_milliseconds`

### Database encoding requirements

The PostgreSQL database must now use the `UTF8` encoding. This is the default encoding that PostgreSQL uses. Unless you have specifically chosen a different encoding when creating the authentik database, no change is needed.

## New features and improvements

- **LDAP Provider improvements**:

>     - the group object returned on queries to the LDAP provider now includes a `memberOf` field pointing back to its parent if set. Contributed by [@danieladugyan](https://github.com/danieladugyan), thank you!

Reviewer 2: ```suggestion
    - Group objects returned by queries to the LDAP provider now includes a `memberOf` field pointing back to its parent if set. Contributed by [@danieladugyan](https://github.com/danieladugyan), thank you!
```

>     - the LDAP connections are now terminated upon session deletion in authentik.

Reviewer 2: ```suggestion
    - LDAP connections are now terminated upon session deletion in authentik.
```

> - **Accessibility improvements**: better screen reader support within the admin interface for navigation, forms, and wizard elements.

Reviewer 2: ```suggestion
- **Accessibility improvements**: Better screen reader support within the admin interface for navigation, forms, and wizard elements.
```

> - **Send event notifications to associated user**: configure notification rules to send the event notification to the user associated to the event. [Read more in our documentation](../../sys-mgmt/events/notifications.md).

Reviewer 2: ```suggestion
- **Send event notifications to associated user**: Configure notification rules to send the event notification to the user associated to the event. Read more in our [notifications documentation](../../sys-mgmt/events/notifications.md).
```
